pkg/gadget-collection/gadgets/trace/network/gadget.go

```diff
@@ -236,39 +238,56 @@ func (t *Trace) run(trace *gadgetv1alpha1.Trace) {
 	traceBeforePatch := trace.DeepCopy()
 	ticker := time.NewTicker(time.Second)
 
+	update := func() bool {
```

Comment:
What about implementing it like a normal method of Tracer. (`func (t *Trace)  update() bool`)?